File: src/investments/registered-education-savings-plan/quebec-education-savings-incentive.ts
import { initializeSavingsGrant, SavingsGrant, SavingsGrantConfig } from './savings-grant';

const quebecEducationSavingsIncentiveConfig: SavingsGrantConfig = {
    MAX_GRANT: 3600,
    MAX_BENEFICIARY_AGE: 17,
    MAX_AMOUNT_YEARLY_FOR_GRANT: 2500,
    YEARLY_GRANT_PERCENT: 0.1,
    MAX_AMOUNT_FOR_SUPP_GRANT: 500,
    SUPP_GRANT_PERCENT: {
        LOW_INCOME: 0.1,
        MEDIUM_INCOME: 0.05,
    },
};

// eslint-disable-next-line max-len
export const QuebecEducationSavingsIncentive: SavingsGrant = initializeSavingsGrant(quebecEducationSavingsIncentiveConfig);

File: src/commonMain/kotlin/com/equisoft/taxca/investments/resp/QuebecEducationSavingsIncentive.kt
package com.equisoft.taxca.investments.resp

val QuebecEducationSavingsIncentive: SavingsGrant = SavingsGrant(
    maxGrant = 3600.0,
    maxBeneficiaryAge = 17,
    maxAmountYearlyForGrant = 2500.0,
    yearlyGrantPercent = 0.1,
    maxAmountForSuppGrant = 500.0,
    suppGrantPercent = SuppGrantPercent(
        lowIncome = 0.1,
        mediumIncome = 0.05,
    ),
)
